baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/IllegalSafeLoggingArgumentTest.java

```diff
@@ -1095,6 +1095,26 @@ public void testFieldAnnotationAssignment() {
                 .doTest();
     }
 
+    @Test
+    public void testThrowableFieldAssignment() {
+        helper().addSourceLines(
+                        "Test.java",
+                        "import com.palantir.logsafe.*;",
+                        "import java.util.*;",
+                        "class Test<OutputT> {",
```

Comment:
This test exposed the issue, but only when the (unused) type parameter is present. `JCTree.type` of the field when `<OutputT>` is present is `java.util.Set` and without it, `JCTree.type` is `java.util.Set<java.lang.Throwable>`!

I've updated the check to both:
1. avoid returning type-info based safety in the case that there's no type info
2. check both the symbol type and tree type for safety info